File: Gost/c#/Gost/Encryptor.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using Functions;
using static Functions.Functions;
using Tuple = Functions.Tuple;

class Encryptor
{
    
    
    public static void Encrypt(string  THE_GOAL_FILE,string MAIN_INFORATION_FILE,string ENCRYPTED_FILE)
        {
            BinaryReader info = new BinaryReader( new FileStream(MAIN_INFORATION_FILE, FileMode.Open));
            FileStream input = new FileStream(THE_GOAL_FILE, FileMode.Open);
            BinaryWriter output = new BinaryWriter(new FileStream(ENCRYPTED_FILE, FileMode.OpenOrCreate));
            
            
            List<uint> keysequence = new List<uint>();
            List<uint> keys = ReadKeys(info);
            uint[,] matrix = ReadMatrix(info);


            foreach (uint key in keys)
                output.Write(key);
            for (int i = 0; i < 8; i++)
                for (int j = 0; j < 16; j++)
                    output.Write(matrix[j,i]);


            Console.Write("Count of rounds = ");
            int roundsCount = Convert.ToInt32(Console.ReadLine());
            
            keysequence = GetKeysequence(keys,roundsCount);

            uint fileSize = (uint) input.Length;
            output.Write(fileSize);

            Tuple block = new Tuple();
            for (int i = 0; i < Math.Ceiling(fileSize / 8d); i++)
            {
                block = ReadBlock(input);
                ProcessBlock(block, matrix, keysequence,roundsCount);
                output.Write(block.re);
                output.Write(block.le);
            }
            
            info.Close();
            input.Close();
            output.Close();
        }
}

File: Gost/java/src/Encryptor.java

import nokinobi.functions.Tuple;
import javafx.scene.Scene;
import nokinobi.stream.LittleEndianInputStream;
import nokinobi.stream.LittleEndianOutputStream;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.nio.ByteBuffer;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

import static nokinobi.functions.Functools.*;


public class Encryptor {


    public static void encrypt(String mainInformationFile,String theGoalFile,String encryptedFile) throws Exception {

        File inpitFile = new File(theGoalFile);

        LittleEndianInputStream info = new LittleEndianInputStream(new FileInputStream(mainInformationFile));
        LittleEndianInputStream input= new LittleEndianInputStream(new FileInputStream(theGoalFile));
        LittleEndianOutputStream output= new LittleEndianOutputStream(new FileOutputStream(encryptedFile));

        int[] keys = readKeys(info);
        int[][] matrix = readMatrix(info);

        for(int i=0;i<8;++i)
            output.write(keys[i]);
        for(int j =0; j<8 ; ++j)
            for(int i=0;i<16;++i)
                output.write(matrix[i][j]);

        int file_size = (int) inpitFile.length();
        output.write(file_size);

        System.out.println("Count of rounds = ");
        Scanner scanner = new Scanner(System.in);
        int rounds =  scanner.nextInt();
        int[] keysequence = getKeysequence(keys,rounds);

        Tuple block;
        for (int i = 0; i < Math.ceil(file_size / 8d); ++i) {
            block = readBlock(input);
            processBlock(block, matrix, keysequence);
            output.write(block.re);
            output.write(block.le);
        }

        info.close();
        output.close();
        input.close();
    }

}

